mlir/lib/Dialect/QCO/Transforms/Optimizations/QuantumArgumentPromotion.cpp

```
@@ -0,0 +1,389 @@
+
+/**
+ * @brief Promote tensor arguments to scalar qubits across the whole module.
+ *
+ * @details
+ * Externally visible functions and declarations are skipped because their
+ * signature cannot be changed. At most one argument per function is promoted
+ * per run, because promoting shifts the indices of the remaining arguments.
+ *
+ * @param moduleOp The module to transform.
+ */
+void runQuantumArgumentPromotion(ModuleOp moduleOp) {
```

Cppcheck (2.21.0)

[style] The function 'runQuantumArgumentPromotion' is never used.

(unusedFunction)